cpp/include/cpu_operations.h

```diff
@@ -48,8 +48,35 @@ class CpuOperations {
     // Matrix-matrix multiplication
     return a * b;
   }
-  static Matrix<T> Add(const Matrix<T> &a, const T &scalar);
-  static Matrix<T> Add(const Matrix<T> &a, const Matrix<T> &b);
+  static Matrix<T> Add(const Matrix<T> &a, const T &scalar) {
+      // Does not work if matrix is empty.
+      if (a.rows() == 0) {
+        std::cerr << "MATRICIES ARE EMPTY";
+        exit(1);
+
+      // Otherwise, code will run fine.
+    } else {
```

Comment:
here, use array() in Eigen to add scalar to a matrix, Jason posted the link in the campfire, also please let Anne know about this because she is working on scalar matrix subtraction as well.